x/leverage/keeper/borrows.go

```diff
@@ -90,7 +132,7 @@ func (k Keeper) CalculateBorrowLimit(ctx sdk.Context, collateral sdk.Coins) (sdk
 		// ignore blacklisted tokens
 		if !ts.Blacklist {
 			// get USD value of base assets
-			v, err := k.TokenValue(ctx, baseAsset)
+			v, err := k.TokenValue(ctx, baseAsset, historic)
```

Comment:
They are all good (mostly because the checks are at the end of each message) except for `MsgMaxWithdraw` which does the (query-like) `maxWithdraw` first but still checks health at the end.

I was considering ways to streamline that, but in the case of `MsgMaxWithdraw` it would require a lot of spaghetti code to save that gas.